include/ginkgo/core/base/device_matrix_data.hpp

```diff
@@ -94,13 +125,11 @@ struct device_matrix_data {
      *
      * @param exec  the executor to create the device_matrix_data on.
      * @param data  the data to be wrapped or copied into a device_matrix_data.
-     * @return  a device_matrix_data object with the same size as `data` and the
-     *          same entries, either wrapped as a non-owning view if `exec` is a
-     *          host executor or copied into an owning Array if `exec` is a
-     *          device executor.
+     * @return  a device_matrix_data object with the same size and entries as
+     *          `data` copied to the device executor.
      */
-    static device_matrix_data create_view_from_host(
-        std::shared_ptr<const Executor> exec, host_type& data);
+    static device_matrix_data create_from_host(
```

Comment:
I can't put in my suggestion directly, since the comment I would like to change is outside your changes. Still, I would change the following line above:
```
Creates a view of the given host data on the given executor.
```
to something like
```
Creates a device_matrix_data from the given host data on the given executor.
```